src/design-system/components/user-avatar/UserAvatar.tsx

```diff
@@ -0,0 +1,106 @@
+import '~scss-components/user-avatar/UserAvatar.scss'
+import { forwardRef } from 'react'
+import {
+  Avatar as MuiAvatar,
+  AvatarProps as MuiAvatarProps
+} from '@mui/material'
+import CheckIcon from '@mui/icons-material/Check'
+import { cn } from '~/utils/cn'
+
+const variants = ['check', 'avatar', 'monogram', 'photo'] as const
+const sizes = ['sm', 'md', 'lg'] as const
+
+interface BaseUserAvatarProps {
+  variant?: (typeof variants)[number]
+  src?: string
+  size?: (typeof sizes)[number]
+  firstName: string
+  lastName: string
+  isOnline?: boolean
+}
+
+export type UserAvatarProps = BaseUserAvatarProps &
+  Omit<MuiAvatarProps, keyof BaseUserAvatarProps>
+
+type Ref = MuiAvatarProps['ref']
+const UserAvatar = forwardRef(
+  (
+    {
+      variant = 'avatar',
+      src,
+      size = 'sm',
+      firstName,
+      lastName,
+      isOnline,
+      onClick,
+      ...props
+    }: UserAvatarProps,
+    forwardedRef: Ref
+  ) => {
```

Comment:
It is better to use generic in `forwardRef` function. You will not have to type props and forwardedRef manually.
```suggestion
const UserAvatar = forwardRef<HTMLDivElement, UserAvatarProps>(
  (
    {
      variant = 'avatar',
      src,
      size = 'sm',
      firstName,
      lastName,
      isOnline,
      onClick,
      ...props
    },
   ref
  ) => {
```

Reply:
![image](https://github.com/user-attachments/assets/fdbb725b-d999-4de8-bb21-952a454bbe4b)
the DOM element type is HTMLDivElement at all levels, so I don't see the point in specifically specifying HTMLDivElement for ref

Reply:
You can keep the ref, but it's better to extract it into a generic `forwardRef<Ref, UserAvatarProps`.